sympy/polys/rootisolation.py

```diff
@@ -772,7 +772,7 @@ def _real_isolate_and_disjoin(factors, K, eps=None, inf=None, sup=None, strict=F
         I_pos = [ (( u, v), k) for ((u, v), k, _) in I_pos ]
     else:
         I_neg = [ ((-v, -u), k, f) for ((u, v), k, f) in I_neg ]
-        I_pos = [ (( u, v), k, f) for ((u, v), k, f) in I_pos ]
+        I_pos = I_pos
```

Comment:
I don't always agree with these changes. It can often be clearer to see the four lines of code against each other even if it means that some lines involve redundant calculations. I doubt very much that this contributes significantly to runtime.

In any case I would simplify it like:
```python
I_neg = [((-v, -u), k, f) for ((u, v), k, f) in I_neg]

if not basis:
    I_neg = [((u, v), k) for ((u, v), k, _) in I_neg]
    I_pos = [((u, v), k) for ((u, v), k, _) in I_pos]
```
That way the control flow is clearer because the effect of `basis` is separated from the unrelated need to negate the values in `I_neg`.